File: Sorting/selection/selection.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

// selectionSort sorts a slice of integers in-place.
func selectionSort(nums []int) {
	n := len(nums)
	for i := 0; i < n-1; i++ {
		// Find the minimum element in the unsorted part of the slice
		minIndex := i
		for j := i + 1; j < n; j++ {
			if nums[j] < nums[minIndex] {
				minIndex = j
			}
		}
		// Swap the found minimum with the first element of the unsorted part
		nums[i], nums[minIndex] = nums[minIndex], nums[i]
	}
}

func main() {
	// 1. Set up the reader
	reader := bufio.NewReader(os.Stdin)
	fmt.Println("Enter numbers separated by spaces (e.g., 64 25 12 22 11):")
	
	line, _ := reader.ReadString('\n')
	line = strings.TrimSpace(line)

	// 2. Parse the input string into a slice of integers
	stringNumbers := strings.Split(line, " ")
	numbers := make([]int, 0, len(stringNumbers)) // Initialize with capacity for efficiency

	for _, s := range stringNumbers {
		num, err := strconv.Atoi(s)
		if err != nil {
			fmt.Println("Error: Invalid input. Please enter only integers.")
			return
		}
		numbers = append(numbers, num)
	}

	// 3. Call the sorting function
	selectionSort(numbers)

	// 4. Print the result
	fmt.Println("Sorted Array:", numbers)
}

File: Sorting/selection/selection.py
from typing import List

def selection_sort(nums: List[int]) -> None:
    """
    Sorts a list of integers in-place using the Selection Sort algorithm.
    """
    n = len(nums)
    # Traverse through all array elements
    for i in range(n - 1):
        # Find the minimum element in the remaining unsorted array
        min_index = i
        for j in range(i + 1, n):
            if nums[j] < nums[min_index]:
                min_index = j

        # Swap the found minimum element with the first element of the unsorted part
        # using Python's tuple unpacking for a clean swap.
        nums[i], nums[min_index] = nums[min_index], nums[i]

def main():
    """
    Main function to run the program.
    """
    # 1. Get input from the user
    print("Enter numbers separated by spaces (e.g., 64 25 12 22 11):")
    line = input()

    # 2. Parse the string into a list of integers
    try:
        numbers = [int(s) for s in line.split()]
    except ValueError:
        print("Invalid input. Please enter only integers.")
        return

    # 3. Call the sorting function
    selection_sort(numbers)

    # 4. Print the sorted list
    print(f"Sorted Array: {numbers}")

if __name__ == "__main__":
    main()